### PR TITLE
chore(deps): remove helm as a dependency

### PR DESCRIPTION
**Pull request by ecordell**
**Description of the change:**
Removes helm as a dependency

**Motivation for the change:**
This is purely to unblock vendoring in `oc` - helm brings in many dependencies that we do not currently use and which cause conflicts with oc.

Eventually we may want to pull helm in, but for now I have vendored just the bits for identifying a chart.

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: pkg/lib/bundle/chartutil.go [hunk 1]
```diff
@@ -1,10 +1,11 @@
 /*
 Copyright The Helm Authors.
+
 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at
 
-http://www.apache.org/licenses/LICENSE-2.0
+    http://www.apache.org/licenses/LICENSE-2.0
```
Comment (dinhxuanvu) [inline pkg/lib/bundle/chartutil.go]: nit: this is not important but would be nice to have it look prettier. 😄

File: pkg/lib/bundle/chartutil.go [hunk 2]
```diff
@@ -64,31 +108,38 @@ type Metadata struct {
...
-	if md.Version == "" {
-		return ValidationError("chart.metadata.version is required")
+
+	chartContent := new(Metadata)
+	if err := yaml.Unmarshal(chartYamlContent, &chartContent); err != nil {
+		return false, err
 	}
-	if !isValidChartType(md.Type) {
-		return ValidationError("chart.metadata.type must be application or library")
+	if chartContent == nil {
```
Comment (benluddy) [inline pkg/lib/bundle/chartutil.go]: Does this ever evaluate to true?

**Pull request merged**